src/components/motor/MotorController.cpp

```diff
@@ -13,19 +14,43 @@ MotorController::MotorController(Controllers::Settings& settingsController) : se
 void MotorController::Init() {
   nrf_gpio_cfg_output(pinMotor);
   nrf_gpio_pin_set(pinMotor);
-  app_timer_create(&vibTimer, APP_TIMER_MODE_SINGLE_SHOT, vibrate);
+  app_timer_init();
+  
+  app_timer_create(&shortVibTimer, APP_TIMER_MODE_SINGLE_SHOT, vibrate);
+  app_timer_create(&longVibTimer, APP_TIMER_MODE_REPEATED, vibrate);
+  isBusy = false;
 }
 
-void MotorController::SetDuration(uint8_t motorDuration) {
+void MotorController::runForDuration(uint8_t motorDuration) {
 
-  if (settingsController.GetVibrationStatus() == Controllers::Settings::Vibration::OFF)
+  if (settingsController.GetVibrationStatus() == Controllers::Settings::Vibration::OFF || isBusy)
     return;
 
   nrf_gpio_pin_clear(pinMotor);
   /* Start timer for motorDuration miliseconds and timer triggers vibrate() when it finishes*/
-  app_timer_start(vibTimer, APP_TIMER_TICKS(motorDuration), NULL);
+  app_timer_start(shortVibTimer, APP_TIMER_TICKS(motorDuration), NULL);
 }
 
-void MotorController::vibrate(void* p_context) {
+void MotorController::startRunning(uint8_t motorDuration) {
+  if (settingsController.GetVibrationStatus() == Controllers::Settings::Vibration::OFF || isBusy )
+    return;
+  //prevent other vibrations while running
+  isBusy = true;
+  nrf_gpio_pin_clear(pinMotor);
+  app_timer_start(longVibTimer, APP_TIMER_TICKS(motorDuration), NULL);
+}
+
+void MotorController::stopRunning() {
+  
+  app_timer_stop(longVibTimer);
   nrf_gpio_pin_set(pinMotor);
+  isBusy = false;
+}
+
+void MotorController::vibrate(void* p_context) {
+  if (nrf_gpio_pin_out_read(pinMotor) == 0) {
+    nrf_gpio_pin_set(pinMotor);
+  } else {
+    nrf_gpio_pin_clear(pinMotor);
+  }
```

Comment:
Maybe this function should play a pattern on each call. This would work for both a short and continuous notification and make them more noticable, and nicer. Just two short vibrations could work.

Reply:
Vibration patterns would be a nice addition to this class `MotorController`. However, I think we can create a new PR to add them :)

Reply:
I was looking at extending the `MotorController` interface with something like `void vibrate(std::chrono::milliseconds pulseDuration, int pulseCount = 1, std::chrono::milliseconds = some_reasonable_default)` (or maybe a 0-100 duty cycle as the last parameter, defaulted to 50). If you think something like that would be useful here I can keep working on it.

Reply:
I think that the current 50% duty cycle is too aggressive and uncomfortable for the user. We should also limit the maximum duration the motor is running, because even in Metronome we can hear it get weaker.